ci/tasks/terraform_apply.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# Runs a `terraform apply` on ${TERRAFORM_SOURCE}
+
+set -euo pipefail
+
+: ${WORKSPACE}
+: ${TERRAFORM_SOURCE}
+: ${TF_VAR_BPM_USER}
+: ${TF_VAR_BPM_PW}
+: ${AWS_REGION}
+: ${S3_NAME}
+: ${S3_KEY}
+
+# Ensure the workspace doesn't have any invalid character
+if [[ ${WORKSPACE} =~ - ]]; then
+    echo "Terraform workspace cannot contain '-' (breaks some resource names)"
+    exit 1
+fi
+
+. ./bpm-mail-lambda/ci/tasks/util/assume_role.sh
+. ./bpm-mail-lambda/ci/tasks/util/setup_terraform.sh
+
+terraform apply \
+    --auto-approve
```

Comment:
Don't you need to also pass the variables BPM_UESR and BPM_PW into the terraform here as those two don't have defaults unless it's picking it up from elsewhere.

Reply:
No, they are pulled from the pipeline.yml in tf_params